storage/remote/store.go

```diff
@@ -176,11 +224,6 @@ func (s *Store) LedgerByHeight(
 	ctx context.Context,
 	blockHeight uint64,
 ) (snapshot.StorageSnapshot, error) {
-	err := s.SetBlockHeight(blockHeight)
-	if err != nil {
-		return nil, err
-	}
-
 	return snapshot.NewReadFuncStorageSnapshot(func(id flowgo.RegisterID) (flowgo.RegisterValue, error) {
 		// first try to see if we have local stored ledger
 		value, err := s.DefaultStore.GetBytesAtVersion(
```

Comment:
I think here needs a check, something like: 

```go
if blockHeight <= s.CurrentHeight{
    value, err := s.DefaultStore.GetBytesAtVersion(
        ctx,
        s.KeyGenerator.Storage(storage.LedgerStoreName),
        []byte(id.String()),
        blockHeight,
	)
}

```

Reply:
great point. I think this should just rewrite the height if it's greater than the fork height